### PR TITLE
Add pointer methods

### PR DESCRIPTION
**Pull request by mwielbut**
This PR adds some convenience methods to make it easier to convert between pointer and Optional representation of values. 

Using the Optional package inevitably leads to lots of code like this when we need to convert back to pointers:
```
a.FirstName.If(func(v string) {
	row.FirstName = &v
})
```

or the other way:

```
if r.FirstName != nil {
	a.FirstName = optional.NewString(*r.FirstName)
}
```

Two new functions are introduced, one to create an Optional given a pointer, and the other to return the embedded value as a pointer. Note, in both cases the value is copied over, the pointer is never passed directly as that would break the encapsulation. 

With the two methods, the above examples become simple one-line statements:

```
row.FirstName = a.FirstName.ToPointer()
```

```
a.FirstName = optional.NewStringFromPointer(r.FirstName)
```

### CODE REVIEW

File: cmd/optional/main.go [hunk 1]
```diff
@@ -146,11 +146,28 @@ func New{{ .OutputName }}(v {{ .TypeName }}) {{ .OutputName }} {
 	return {{ .OutputName }}{&v}
 }
 
+// New{{ .OutputName }}FromPointer creates an optional.{{ .OutputName }} from a {{ .TypeName }} pointer.
```
Comment (markphelps) [inline cmd/optional/main.go:149]: Better late than never ;)

WDYT about `FromPtr` / `ToPtr` ? saves some keystrokes and I think is just as readable?

Comment (mwielbut) [reply cmd/optional/main.go:149]: I like it. I’ll push up and update

Comment (mwielbut) [reply cmd/optional/main.go]: Updated 👍

**Review by markphelps — APPROVED** [review]
Lgtm! Thank you. Will create a new release in the morning

**Comment by mwielbut** [comment]
> Lgtm! Thank you. Will create a new release in the morning

@markphelps any chance we could deploy today? I'd love to update our code to use this :)

**Comment by markphelps** [comment]
@mwielbut thanks again for the contribution!!

I just created https://github.com/markphelps/optional/releases/tag/v0.11.0 with the change

I'm going to be paying more attention to this library going forward and hope to do more regular releases. Thanks again!

**Pull request merged**